### PR TITLE
Remove deprecation alerts from post editors

### PR DESCRIPTION
**Pull request by SergioEstevao**
Update all the code that use UIAlertView, UIActionSheet and UIPopover to be compatible iOS 9 code.

As a side effect this code also makes things work better on iPad on multitask mode and iPhone Plus on landscape mode.

In order to test it some scenarios that use these views need to be reviewed on regular iPhone, iPhone Plus and iPad

- Edit post and click on the X button to see the discard alert/actions
- Add images and/or video and click on them while the upload is happening
- Click on the blog selector
- While uploads are going click on the HTML switch or try to close the editor.

Needs Review: @jleandroperez, thanks in advance!

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Post/WPLegacyEditPostViewController.m [hunk 1]
```diff
@@ -244,30 +235,25 @@ - (void)showBlogSelectorPrompt
...
-                                              otherButtonTitles:NSLocalizedString(@"OK",@""), nil];
-    alertView.tag = EditPostViewControllerAlertTagSwitchBlogs;
-    [alertView show];
+    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:NSLocalizedString(@"Change Site", @"Title of an alert prompting the user that they are about to change the blog they are posting to.")
+                                                                             message:NSLocalizedString(@"Choosing a different site will lose edits to site specific content like media and categories. Are you sure?", @"And alert message warning the user they will loose blog specific edits like categories, and media if they change the blog being posted to.")
+                                                                      preferredStyle:UIAlertControllerStyleAlert];
+    [alertController addDefaultActionWithTitle:NSLocalizedString(@"OK",@"") handler:^(UIAlertAction *action) {
+        [self showBlogSelector];
+    }];
+    [alertController addCancelActionWithTitle:NSLocalizedString(@"Cancel",@"") handler:^(UIAlertAction *action) {
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Post/WPLegacyEditPostViewController.m]: The handler is optional, you can just send `nil` in there

File: WordPress/Classes/ViewRelated/Post/WPLegacyEditPostViewController.m [hunk 2]
```diff
@@ -244,30 +235,25 @@ - (void)showBlogSelectorPrompt
...
-    }
-
     void (^dismissHandler)() = ^(void) {
-        if (IS_IPAD) {
-            [self.blogSelectorPopover dismissPopoverAnimated:YES];
-        } else {
-            self.dismissingBlogPicker = YES;
-            [self dismissViewControllerAnimated:YES completion:nil];
-            self.dismissingBlogPicker = NO;
-        }
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Post/WPLegacyEditPostViewController.m]: *Beautiful* cleanup Sergio thank you!

File: WordPress/Classes/ViewRelated/Post/WPPostViewController.m [hunk 3]
```diff
@@ -656,30 +630,22 @@ - (void)showBlogSelectorPrompt:(WPBlogSelectorButton*)sender
...
-        if (IS_IPAD) {
-            [self.blogSelectorPopover dismissPopoverAnimated:YES];
-        } else {
-            self.dismissingBlogPicker = YES;
-            [self dismissViewControllerAnimated:YES completion:nil];
-            self.dismissingBlogPicker = NO;
-        }
+        self.dismissingBlogPicker = YES;
+        [self dismissViewControllerAnimated:YES completion:nil];
+        self.dismissingBlogPicker = NO;
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Post/WPPostViewController.m]: You can kill the `self.dismissingBlogPicker` property. Just checked, it's not being used anywhere

File: WordPress/Classes/ViewRelated/Post/WPPostViewController.m [hunk 4]
```diff
@@ -730,19 +696,11 @@ - (void)showBlogSelector
     UINavigationController *navController = [[UINavigationController alloc] initWithRootViewController:vc];
     navController.navigationBar.translucent = NO;
     navController.navigationBar.barStyle = UIBarStyleBlack;
-    
-    if (IS_IPAD) {
-        vc.preferredContentSize = CGSizeMake(320.0, 500);
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Post/WPPostViewController.m]: Die, magic number, die! :smile:

File: WordPress/Classes/ViewRelated/Post/WPPostViewController.m [hunk 5]
```diff
@@ -854,46 +813,49 @@ - (void)cancelEditing
...
-- (void)showPostHasChangesActionSheet
+- (void)showPostHasChangesAlert
 {
-	UIActionSheet *actionSheet;
+    UIAlertController *alertController;
+    alertController = [UIAlertController alertControllerWithTitle:NSLocalizedString(@"You have unsaved changes.", @"Title of message with options that shown when there are unsaved changes and the author is trying to move away from the post.")
+                                                          message:nil
+                                                   preferredStyle:UIAlertControllerStyleActionSheet];
+    [alertController addActionWithTitle:NSLocalizedString(@"Keep Editing", @"Button shown if there are unsaved changes and the author is trying to move away from the post.")
+                                  style:UIAlertActionStyleCancel
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Post/WPPostViewController.m]: It's super cool to see that... `UIAlertController`, on iPad devices, will automatically hide cancel actions

**Comment by jleandroperez** [comment]
@SergioEstevao looking great!

Added few minor code comments. Once addressed, please, feel free to :shipit: !

**Comment by SergioEstevao** [comment]
Thanks @jleandroperez

**Pull request merged**